### PR TITLE
Tags used by dogstats

### PR DESCRIPTION
**Pull request by shmish111**
Not sure what you think about this. I had to fork your repo so that I could use 'tags' in dogstats. It's an extension to statsd so in a way it doesn't fit into your library, then again it seems a bit pointless having a fork.

### CODE REVIEW

File: src/clj_statsd.clj [hunk 1]
```diff
@@ -46,43 +47,50 @@
 (defn publish
```
Comment (pyr) [inline src/clj_statsd.clj]: Can you make publish a 3 and 4 arity function, with a default value of `[]` for tags?

File: project.clj [hunk 2]
```diff
@@ -1,4 +1,6 @@
-(defproject clj-statsd "0.3.11"
+(defproject shmish111/clj-statsd "0.4.0"
```
Comment (pyr) [inline project.clj]: Please keep the artifact description as previously. I'm fine with bumping to 0.4.0

**Comment by pyr** [comment]
I there, thanks for the PR!
I have minor nitpicks, otherwise fine.

**Comment by shmish111** [comment]
I updated `publish`, is this what you meant? I also added some tests

File: src/clj_statsd.clj [hunk 3]
```diff
@@ -46,43 +47,52 @@
...
+  ([^String content rate]
+   (publish content rate []))
+  ([^String content rate tags]
+   (let [prefix (:prefix @cfg)
+         with-prefix (if prefix (str prefix content) content)
+         with-rate (cond
+                     (nil? @cfg) nil
+                     (>= rate 1.0) with-prefix
+                     (<= (.nextDouble ^Random (:random @cfg)) rate) (format "%s|@%f" with-prefix rate))
+         with-tags (if (not-empty tags) (format "%s|#%s" with-rate (str/join "," tags)) with-rate)]
```
Comment (pyr) [inline src/clj_statsd.clj]: `not-empty` should be `seq`

**Comment by pyr** [comment]
Thanks!

**Comment by shmish111** [comment]
Ditto! Can you let me know when you release it to Clojars?

**Comment by pyr** [comment]
@shmish111 git tag `0.4.0` corresponds to the `0.4.0` release on clojars, I added a small refactor

**Pull request merged**